src/main/java/de/triology/cas/logout/LogoutUriEnabledLogoutManagerImpl.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package de.triology.cas.logout;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.Deflater;
+
+import javax.validation.constraints.NotNull;
+
+import de.triology.cas.services.LogoutUriEnabledRegexRegisteredService;
+import org.apache.commons.codec.binary.Base64;
+import org.jasig.cas.authentication.principal.Service;
+import org.jasig.cas.authentication.principal.SingleLogoutService;
+import org.jasig.cas.logout.LogoutMessageCreator;
+import org.jasig.cas.logout.LogoutRequest;
+import org.jasig.cas.logout.LogoutRequestStatus;
+import org.jasig.cas.services.LogoutType;
+import org.jasig.cas.services.RegisteredService;
+import org.jasig.cas.services.ServicesManager;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.jasig.cas.util.HttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This logout manager handles the Single Log Out process.
+ *
+ * @author Jerome Leleu
+ * @since 4.0.0
+ */
+public final class LogoutUriEnabledLogoutManagerImpl implements org.jasig.cas.logout.LogoutManager {
+
+    /** The logger. */
+    private static final Logger LOGGER = LoggerFactory.getLogger(LogoutUriEnabledLogoutManagerImpl.class);
+
+    /** ASCII character set. */
+    private static final Charset ASCII = Charset.forName("ASCII");
+
+    /** The services manager. */
+    @NotNull
+    private final ServicesManager servicesManager;
+
+    /** An HTTP client. */
+    @NotNull
+    private final HttpClient httpClient;
+
+    @NotNull
+    private final LogoutMessageCreator logoutMessageBuilder;
+
+    /** Whether single sign out is disabled or not. */
+    private boolean disableSingleSignOut = false;
+
+    /**
+     * Build the logout manager.
+     * @param servicesManager the services manager.
+     * @param httpClient an HTTP client.
+     * @param logoutMessageBuilder the builder to construct logout messages.
+     */
+    public LogoutUriEnabledLogoutManagerImpl(final ServicesManager servicesManager, final HttpClient httpClient,
+                                             final LogoutMessageCreator logoutMessageBuilder) {
+        this.servicesManager = servicesManager;
+        this.httpClient = httpClient;
+        this.logoutMessageBuilder = logoutMessageBuilder;
+    }
+
+    /**
+     * Perform a back channel logout for a given ticket granting ticket and returns all the logout requests.
+     *
+     * @param ticket a given ticket granting ticket.
+     * @return all logout requests.
+     */
+    @Override
+    public List<LogoutRequest> performLogout(final TicketGrantingTicket ticket) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "performLogout" is 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)